docs/targets/solana.rst

```diff
@@ -23,60 +23,96 @@ This is how to build your Solidity for Solana:
+        const programId = new anchor.web3.PublicKey(IDL.metadata.address);
 
-        const program = Keypair.generate();
-        const storage = Keypair.generate();
+        const wallet = provider.wallet.publicKey;
 
-        const contract = new Contract(connection, program.publicKey, storage.publicKey, FLIPPER_ABI, payer);
+        const program = new anchor.Program(IDL, programId, provider);
 
-        await contract.load(program, PROGRAM_SO);
+        await program.methods.new(wallet, true)
+            .accounts({ dataAccount: dataAccount.publicKey })
+            .signers([dataAccount]).rpc();
 
-        console.log('Program deployment finished, deploying the flipper contract ...');
+        const val1 = await program.methods.get()
+            .accounts({ dataAccount: dataAccount.publicKey })
+            .view();
 
-        await contract.deploy('flipper', [true], storage, 17);
+        console.log(`state: ${val1}`);
 
-        const res = await contract.functions.get();
-        console.log('state: ' + res.result);
+        await program.methods.flip()
+            .accounts({ dataAccount: dataAccount.publicKey })
+            .rpc();
 
-        await contract.functions.flip();
+        const val2 = await program.methods.get()
+            .accounts({ dataAccount: dataAccount.publicKey })
+            .view();
 
-        const res2 = await contract.functions.get();
-        console.log('state: ' + res2.result);
+        console.log(`state: ${val2}`);
     })();
 
-The contract can be used via the `@solana/solidity <https://www.npmjs.com/package/@solana/solidity>`_  npm package. This
-package has `documentation <https://solana-labs.github.io/solana-solidity.js/>`_ and there
-are `some examples <https://solana-labs.github.io/solana-solidity.js/>`_. There is also
+Now you'll have to set the `ANCHOR_WALLET` and `ANCHOR_PROVIDER_URL` environment variables to the correct values in order to run the example.
+
+.. code-block:: bash
+
+    export ANCHOR_WALLET=$HOME/.config/solana/id.json
+    export ANCHOR_PROVIDER_URL=http://127.0.0.1:8899
+    node flipper.js
+
+For more examples, see the
 `solang's integration tests <https://github.com/hyperledger/solang/tree/main/integration/solana>`_.
 
+Using the Anchor client library
+_______________________________
+
+Some notes on using the anchor javascript npm library.
+
+* Solidity function names are converted to camelCase. This means that if in Solidity a function is called ``foo_bar()``,
+  then you must write ``fooBar()`` in your javascript.
+* Anchor only allows you to call ``.view()`` on Solidity functions which are declared ``view`` or ``pure``.
+* Named return values in Solidity are also converted to camelCase. Unnamed returned are given the name ``return0``, ``return1``, etc,
+  depending on the position in the returns values.
+* Return values from non-view functions are not decoded and are not accessible. This is a limitation in the
+  Anchor library. Possibly this can be fixed.
```

Comment:
By checking the pure functions from `builtins.sol` test, this comment should exclude pure functions too (they are `non-view`).

Reply:
fixed